resilience4j-spring-boot-common/src/main/java/io/github/resilience4j/springboot/common/circuitbreaker/autoconfigure/AbstractCircuitBreakerConfigurationOnMissingBean.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2019 Mahmoud Romeh
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.github.resilience4j.springboot.common.circuitbreaker.autoconfigure;
+
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Conditional;
+import org.springframework.context.annotation.Configuration;
+
+import io.github.resilience4j.circuitbreaker.CircuitBreaker;
+import io.github.resilience4j.circuitbreaker.CircuitBreakerRegistry;
+import io.github.resilience4j.circuitbreaker.configure.CircuitBreakerAspect;
+import io.github.resilience4j.circuitbreaker.configure.CircuitBreakerAspectExt;
+import io.github.resilience4j.circuitbreaker.configure.CircuitBreakerConfiguration;
+import io.github.resilience4j.circuitbreaker.configure.CircuitBreakerConfigurationProperties;
+import io.github.resilience4j.circuitbreaker.configure.ReactorCircuitBreakerAspectExt;
+import io.github.resilience4j.circuitbreaker.configure.RxJava2CircuitBreakerAspectExt;
+import io.github.resilience4j.circuitbreaker.event.CircuitBreakerEvent;
+import io.github.resilience4j.consumer.EventConsumerRegistry;
+import io.github.resilience4j.utils.ReactorOnClasspathCondition;
+import io.github.resilience4j.utils.RxJava2OnClasspathCondition;
+
+@Configuration
+public abstract class AbstractCircuitBreakerConfigurationOnMissingBean {
+
+	protected final CircuitBreakerConfiguration circuitBreakerConfiguration;
+	protected final CircuitBreakerConfigurationProperties circuitBreakerProperties;
+
+	public AbstractCircuitBreakerConfigurationOnMissingBean(CircuitBreakerConfigurationProperties circuitBreakerProperties) {
+		this.circuitBreakerProperties = circuitBreakerProperties;
+		this.circuitBreakerConfiguration = new CircuitBreakerConfiguration(circuitBreakerProperties);
+	}
+
+	@Bean
+	@ConditionalOnMissingBean
+	public CircuitBreakerRegistry circuitBreakerRegistry(EventConsumerRegistry<CircuitBreakerEvent> eventConsumerRegistry) {
+		CircuitBreakerRegistry circuitBreakerRegistry = CircuitBreakerRegistry.ofDefaults();
```

Comment:
Could we initialize a Registry like this?

```
Map<String, CircuitBreakerConfig> configs = circuitBreakerProperties.getConfigs()
    .entrySet().stream().collect(Collectors.toMap(Map.Entry::getKey,
						entry -> circuitBreakerProperties.createCircuitBreakerConfig(entry.getValue())));
CircuitBreakerRegistry circuitBreakerRegistry = CircuitBreakerRegistry.of(configs);
```

If the Map contains no "default" key, then we do in the constructor of InMemoryCircuitBreakerRegistry.
```CircuitBreakerConfig defaultConfiguration = configs.getOrDefault("default", CircuitBreakerConfig.ofDefaults());```